src/LuaTransformer.ts

```diff
@@ -4557,10 +4621,10 @@ export class LuaTransformer {
             if (symbols.some(s => this.shouldHoist(s, scope))) {
                 let assignment: tstl.AssignmentStatement | undefined;
                 if (declaration.right) {
-                    assignment = tstl.createAssignmentStatement(
-                        declaration.left,
-                        declaration.right
-                    );
+                    assignment = tstl.createAssignmentStatement(declaration.left, declaration.right);
+                    // Preserve position info for sourcemap
```

Comment:
Probably cleaner if you use `tstl.setNodeOriginal` here, instead of manually setting line and column.

Reply:
The original TS node is gone at this point. I'd have to restructure things to preserve that.